db/types/operations/cast.py

```diff
@@ -1028,6 +1063,91 @@ def _build_numeric_array_function():
     """
 
 
+def _get_text_to_integer_cast():
+    text_db_type_id = PostgresType.TEXT.id
+    integer_db_type_id = PostgresType.BIGINT.id
+
+    integer_array_function = get_qualified_name(INTEGER_ARR_FUNC_NAME)
+    cast_exception_str = (
+        f"RAISE EXCEPTION '% cannot be cast to {PostgresType.BIGINT}', $1;"
+    )
+    return rf"""
+    DECLARE is_negative {PostgresType.BOOLEAN.id};
+    DECLARE integer_arr {text_db_type_id}[];
+    DECLARE integer {text_db_type_id};
+    BEGIN
+        SELECT {integer_array_function}($1::{text_db_type_id}) INTO integer_arr;
+        IF integer_arr IS NULL THEN
+            {cast_exception_str}
+        END IF;
+
+        SELECT integer_arr[1] INTO integer;
+        SELECT $1::text ~ '^-.*$' INTO is_negative;
+
+        IF integer_arr[2] IS NOT NULL THEN
+            SELECT regexp_replace(integer, integer_arr[2], '', 'gq') INTO integer;
+        END IF;
+        IF is_negative THEN
+            RETURN ('-' || integer)::{integer_db_type_id};
+        END IF;
+        RETURN integer::{integer_db_type_id};
+    END;
+    """
+
+
+def _build_integer_array_function():
+    """
+    The main reason for this function to be separate is for testing. This
+    does have some performance impact; we should consider inlining later.
+    """
+    qualified_function_name = get_qualified_name(INTEGER_ARR_FUNC_NAME)
+
+    no_separator = r"\d+"
+    comma_or_period_separator = r"([0-9]|[0-9]{3})(([,.])[0-9]{3})+"
+    comma_separator_lakh_system = r"([0-9]{1,2}(?:(,)[0-9]{2})+),[0-9]{3}"
+    space_separator = r"[0-9]{1,3}(?:( )[0-9]{3})+"
+    single_quote_separator = r"([0-9]{1,3})(?:(\'')[0-9]{3})+(?:(\'')?[0-9]*)?"
```

Comment:
For the comma and period separator ones, we need at least 2 groups of three to be sure that it's an integer. This is crucial in our inference system at the moment. I.e., it's not possible to determine the value of `1,234` without knowing the locale.

Reply:
oh sorry my bad ,changing the regex to 
comma_or_period_separator = r"([0-9]|[0-9]{3})(([,.])[0-9]{3}){2,}" should fix the issue you are referring to,correct ?